include/rocksdb/slice_transform.h

```diff
@@ -60,6 +60,11 @@ class SliceTransform {
   // This is currently not used and remains here for backward compatibility.
   virtual bool InRange(const Slice& /*dst*/) const { return false; }
 
+  // Some SliceTransform will have a full length which can be used to
+  // determine if two keys are consecuitive. Can be disabled by always
+  // returning 0
+  virtual size_t FullLength() const { return 0; }
```

Comment:
As discussed offline. 0 is a valid number here. I suggest we use
`bool FullLength(size_t* len)` and use the return value to indicate whether there is a full length or not.